src/shell/shell_cmds.cpp

```diff
@@ -2532,6 +2492,93 @@ static std::vector<std::string> search_files(const std::string_view query)
 	return files;
 }
 
+static DOS_Filename split_extension(const std::string& fullname)
+{
+	DOS_Filename ret = {};
+	size_t pos       = fullname.rfind('.');
+	ret.name         = fullname.substr(0, pos);
+	++pos;
+	if (pos > 0 && pos < fullname.size()) {
+		ret.ext = fullname.substr(pos);
+	} else {
+		ret.ext = {};
+	}
+	return ret;
+}
+
+static std::string handle_wildcards(const std::string& wildcards,
+                                    const std::string& old_filename)
+{
+	std::string ret = {};
+	for (size_t i = 0; i < wildcards.size(); ++i) {
+		char c    = wildcards[i];
+		bool done = false;
+		switch (c) {
+		case '*':
+			if (i < old_filename.size()) {
+				ret.append(old_filename, i);
+			}
+			done = true;
+			break;
+		case '?':
+			if (i < old_filename.size()) {
+				ret.push_back(old_filename[i]);
+			}
+			break;
+		default: ret.push_back(c);
+		}
+		if (done) {
+			break;
+		}
+	}
+	return ret;
+}
+
+void DOS_Shell::CMD_RENAME(char* args)
+{
+	HELP("RENAME");
+
+	const std::string first_arg  = DOS_Canonicalize(strip_word(args));
+	const std::string second_arg = strip_word(args);
```

Comment:
Can't you use more descriptive names than first and second? Like source and target?

Reply:
Renamed